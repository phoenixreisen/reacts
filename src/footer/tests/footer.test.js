/**
 * @jest-environment jsdom
 */
 import {render, fireEvent, waitFor, screen} from '@testing-library/react';
 import renderer from 'react-test-renderer';
 import Footer from '../footer.r';
 import React from 'react';

 describe('footer should', () => {

    test('render correctly without params', () => {
        const component = renderer.create(
            <Footer />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('render correctly with params', () => {
        const component = renderer.create(
            <Footer
                headline="Custom Footer Headline"
                urls={{
                    FACEBOOK: 'https://facebook.com',
                    PHXWEBSITE: 'https://meinereise.phoenixreisen.com'
                }}
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });
 });