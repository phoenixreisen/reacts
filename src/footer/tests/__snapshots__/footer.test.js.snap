// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer should render correctly 1`] = `
<footer
  className="footer "
>
  <div
    className="footer__media-links"
  >
    <div
      className="mb1"
    >
      <strong>
        Sie finden uns auch auf
      </strong>
    </div>
    <div
      className="footer__media-icons"
    >
      <a
        className="desktop-only"
        href="https://www.phoenixreisen.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen"
      >
        <img
          alt="Phoenix Reisen Website"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://de-de.facebook.com/PHXTV/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf Facebook"
      >
        <img
          alt="Phoenix Reisen TV auf Facebook"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.youtube.com/PhoenixreisenTV"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf YouTube"
      >
        <img
          alt="Phoenix Reisen TV auf YouTube"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.phoenixreisen.tv/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix TV"
      >
        <img
          alt="Phoenix TV"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Instagram"
      >
        <img
          alt="Phoenix Reisen auf Instagram"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/explore/tags/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Impressionen unserer Gäste"
      >
        <img
          alt="Impressionen unserer Gäste - #phoenixreisen"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://twitter.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Twitter"
      >
        <img
          alt="Phoenix Reisen auf Twitter"
          src="IMGPATH"
        />
      </a>
    </div>
  </div>
  <div
    className="mt2 tc footer__text"
  >
    <div>
      <small>
        Wir sind für Sie da
      </small>
    </div>
    <div>
      <strong>
        <a
          href="tel:+4922892600"
          title="Telefonnummer aufrufen"
        >
          +49 (228) 9260-0
        </a>
      </strong>
       oder über 
      <strong>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          rel="noopener noreferrer"
          target="_blank"
          title="Kontaktformular"
        >
          Kontaktformular
        </a>
      </strong>
      <br />
      <small>
        montags bis freitags von 09:00 bis 18:00 Uhr
      </small>
    </div>
  </div>
  <div
    className="wrapper wrapper--large footer__links-wrapper"
  >
    <ul
      className="footer__links noprint"
    >
      <li>
        <a
          href="https://www.phoenixreisen.com/profibereich.html"
          title="zum Reisebüro-/Profibereich"
        >
          <i
            className="fa fa-angle-right"
          />
          Reisebüro-/Profibereich
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/reise-und-geschaeftsbedingungen.html"
          title="zu unseren AGB"
        >
          <i
            className="fa fa-angle-right"
          />
          AGB
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/datenschutzerklaerung.html"
          title="zu unserer Datenschutzerklärung"
        >
          <i
            className="fa fa-angle-right"
          />
          Datenschutzerklärung
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          title="zu unserer Kontaktseite"
        >
          <i
            className="fa fa-angle-right"
          />
          Kontakt
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/nutzungsbedingungen.html"
          title="zu unseren Nutzungsbedingungen"
        >
          <i
            className="fa fa-angle-right"
          />
          Nutzungsbedingungen
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/impressum.html"
          title="zu unserem Impressum"
        >
          <i
            className="fa fa-angle-right"
          />
          Impressum
        </a>
      </li>
    </ul>
    <div
      className="footer__logos"
    >
      <span>
        <img
          src="IMGPATH"
        />
      </span>
      <span>
        <img
          src="IMGPATH"
        />
      </span>
    </div>
  </div>
</footer>
`;

exports[`footer should render correctly with params 1`] = `
<footer
  className="footer "
>
  <div
    className="footer__media-links"
  >
    <div
      className="mb1"
    >
      <strong>
        Custom Footer Headline
      </strong>
    </div>
    <div
      className="footer__media-icons"
    >
      <a
        className="desktop-only"
        href="https://meinereise.phoenixreisen.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen"
      >
        <img
          alt="Phoenix Reisen Website"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://facebook.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf Facebook"
      >
        <img
          alt="Phoenix Reisen TV auf Facebook"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.youtube.com/PhoenixreisenTV"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf YouTube"
      >
        <img
          alt="Phoenix Reisen TV auf YouTube"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.phoenixreisen.tv/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix TV"
      >
        <img
          alt="Phoenix TV"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Instagram"
      >
        <img
          alt="Phoenix Reisen auf Instagram"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/explore/tags/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Impressionen unserer Gäste"
      >
        <img
          alt="Impressionen unserer Gäste - #phoenixreisen"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://twitter.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Twitter"
      >
        <img
          alt="Phoenix Reisen auf Twitter"
          src="IMGPATH"
        />
      </a>
    </div>
  </div>
  <div
    className="mt2 tc footer__text"
  >
    <div>
      <small>
        Wir sind für Sie da
      </small>
    </div>
    <div>
      <strong>
        <a
          href="tel:+4922892600"
          title="Telefonnummer aufrufen"
        >
          +49 (228) 9260-0
        </a>
      </strong>
       oder über 
      <strong>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          rel="noopener noreferrer"
          target="_blank"
          title="Kontaktformular"
        >
          Kontaktformular
        </a>
      </strong>
      <br />
      <small>
        montags bis freitags von 09:00 bis 18:00 Uhr
      </small>
    </div>
  </div>
  <div
    className="wrapper wrapper--large footer__links-wrapper"
  >
    <ul
      className="footer__links noprint"
    >
      <li>
        <a
          href="https://www.phoenixreisen.com/profibereich.html"
          title="zum Reisebüro-/Profibereich"
        >
          <i
            className="fa fa-angle-right"
          />
          Reisebüro-/Profibereich
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/reise-und-geschaeftsbedingungen.html"
          title="zu unseren AGB"
        >
          <i
            className="fa fa-angle-right"
          />
          AGB
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/datenschutzerklaerung.html"
          title="zu unserer Datenschutzerklärung"
        >
          <i
            className="fa fa-angle-right"
          />
          Datenschutzerklärung
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          title="zu unserer Kontaktseite"
        >
          <i
            className="fa fa-angle-right"
          />
          Kontakt
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/nutzungsbedingungen.html"
          title="zu unseren Nutzungsbedingungen"
        >
          <i
            className="fa fa-angle-right"
          />
          Nutzungsbedingungen
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/impressum.html"
          title="zu unserem Impressum"
        >
          <i
            className="fa fa-angle-right"
          />
          Impressum
        </a>
      </li>
    </ul>
    <div
      className="footer__logos"
    >
      <span>
        <img
          src="IMGPATH"
        />
      </span>
      <span>
        <img
          src="IMGPATH"
        />
      </span>
    </div>
  </div>
</footer>
`;

exports[`footer should render correctly without params 1`] = `
<footer
  className="footer "
>
  <div
    className="footer__media-links"
  >
    <div
      className="mb1"
    >
      <strong>
        Sie finden uns auch auf
      </strong>
    </div>
    <div
      className="footer__media-icons"
    >
      <a
        className="desktop-only"
        href="https://www.phoenixreisen.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen"
      >
        <img
          alt="Phoenix Reisen Website"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://de-de.facebook.com/PHXTV/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf Facebook"
      >
        <img
          alt="Phoenix Reisen TV auf Facebook"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.youtube.com/PhoenixreisenTV"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen TV auf YouTube"
      >
        <img
          alt="Phoenix Reisen TV auf YouTube"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.phoenixreisen.tv/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix TV"
      >
        <img
          alt="Phoenix TV"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Instagram"
      >
        <img
          alt="Phoenix Reisen auf Instagram"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://www.instagram.com/explore/tags/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Impressionen unserer Gäste"
      >
        <img
          alt="Impressionen unserer Gäste - #phoenixreisen"
          src="IMGPATH"
        />
      </a>
      <a
        href="https://twitter.com/phoenixreisen"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen auf Twitter"
      >
        <img
          alt="Phoenix Reisen auf Twitter"
          src="IMGPATH"
        />
      </a>
    </div>
  </div>
  <div
    className="mt2 tc footer__text"
  >
    <div>
      <small>
        Wir sind für Sie da
      </small>
    </div>
    <div>
      <strong>
        <a
          href="tel:+4922892600"
          title="Telefonnummer aufrufen"
        >
          +49 (228) 9260-0
        </a>
      </strong>
       oder über 
      <strong>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          rel="noopener noreferrer"
          target="_blank"
          title="Kontaktformular"
        >
          Kontaktformular
        </a>
      </strong>
      <br />
      <small>
        montags bis freitags von 09:00 bis 18:00 Uhr
      </small>
    </div>
  </div>
  <div
    className="wrapper wrapper--large footer__links-wrapper"
  >
    <ul
      className="footer__links noprint"
    >
      <li>
        <a
          href="https://www.phoenixreisen.com/profibereich.html"
          title="zum Reisebüro-/Profibereich"
        >
          <i
            className="fa fa-angle-right"
          />
          Reisebüro-/Profibereich
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/reise-und-geschaeftsbedingungen.html"
          title="zu unseren AGB"
        >
          <i
            className="fa fa-angle-right"
          />
          AGB
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/datenschutzerklaerung.html"
          title="zu unserer Datenschutzerklärung"
        >
          <i
            className="fa fa-angle-right"
          />
          Datenschutzerklärung
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/kontakt.html"
          title="zu unserer Kontaktseite"
        >
          <i
            className="fa fa-angle-right"
          />
          Kontakt
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/nutzungsbedingungen.html"
          title="zu unseren Nutzungsbedingungen"
        >
          <i
            className="fa fa-angle-right"
          />
          Nutzungsbedingungen
        </a>
      </li>
      <li>
        <a
          href="https://www.phoenixreisen.com/impressum.html"
          title="zu unserem Impressum"
        >
          <i
            className="fa fa-angle-right"
          />
          Impressum
        </a>
      </li>
    </ul>
    <div
      className="footer__logos"
    >
      <span>
        <img
          src="IMGPATH"
        />
      </span>
      <span>
        <img
          src="IMGPATH"
        />
      </span>
    </div>
  </div>
</footer>
`;
